src/main/java/org/kuali/kfs/coa/document/validation/impl/AccountGlobalRule.java

```diff
@@ -863,5 +963,959 @@ private boolean isIncomeStreamAccountRequired(String fundGroupCode, String subFu
         return required;
     }
 
+    protected boolean checkSubFundProgram(MaintenanceDocument document) {
+        boolean success = true;
+
+        String subFundGroupCode = newAccountGlobal.getSubFundGroupCode();
+        String subFundProg = newAccountGlobal.getProgramCode();
+        BusinessObjectService bos = SpringContext.getBean(BusinessObjectService.class);
+
+        if (StringUtils.isNotBlank(subFundProg)) {
+            Map<String, String> fieldValues = new HashMap<String, String>();
```

Comment:
This is an interesting one, after changing it I get this warning:
'Map is a raw type. References to generic type Map<K,V> should be parameterized'